rust/src/dcerpc/dcerpc.rs

```diff
@@ -938,7 +938,7 @@ impl DCERPCState {
 
         let fraglen = self.get_hdr_fraglen().unwrap_or(0);
 
-        if cur_i.len() < (fraglen - frag_bytes_consumed) as usize {
+        if (cur_i.len() + frag_bytes_consumed as usize) < fraglen as usize {
             SCLogDebug!("Possibly fragmented data, waiting for more..");
             return AppLayerResult::incomplete(parsed as u32, fraglen as u32 - parsed as u32);
```

Comment:
The patch looks good but I suppose this part could still give unexpected results if `fraglen` retrieved from the header was <16 as we're telling applayer parser that we need a negative int as u32 more bytes.
Given that the control does reach here, the condition that this could happen in is a broken dcerpc packet that has at least 16 header bytes but `fraglen` has an absurd value <16, I think. wdyt?

Reply:
if fraglen < 16,`(cur_i.len() + frag_bytes_consumed as usize) < fraglen as usize ` cannot be true as `frag_bytes_consumed` is 16

Reply:
indeed true for both incomplete and ok cases. Sorry for the noise.